backend/src/main/java/com/karankumar/bookproject/backend/model/Book.java

```diff
@@ -244,6 +246,20 @@ public BookBuilder edition(Integer edition) {
         }
     }
 
+    public void addPublisher(@NonNull Publisher publisher) {
+            publishers.add(publisher);
+            publisher.getBooks().add(this);
+    }
+
+    public void removePublisher(@NonNull Publisher publisher) {
+        publishers.forEach(bookPublisher -> {
+            if (bookPublisher.getId().equals(publisher.getId())) {
+                bookPublisher.getBooks().remove(this);
```

Comment:
Could you add a `break` here as well? There should only be one match, so we won't need to iterate through the rest of the set (if there are more elements)